### ISSUE TITLE
FPN converter ignoring device

### ISSUE THREAD

**Issue by 23pointsNorth** [chainer/chainercv#812]
Hi all, 
I've been looking through the example FPN training code and came across the [Updater converter](https://github.com/chainer/chainercv/blob/master/examples/fpn/train_multi.py#L96), which is forcing the batch to ignore the device and in essence run on CPU.

When attempted to perform training on a single GPU, I noticed that the model is loaded in the GPU memory, but its utilization is 0%, and there is a CPU core at 100%. As expected, the training is extremely slow. 

If I remove the converter, OpenCV `resize` fails on type (i.e. it's cupy) within the `prepare` call.

Is  this the expected behaviour, or is the code indeed training it only on the CPU?

--Daniel

edit: This is also somewhat also connected to #800, as I had to change the [cast line](https://github.com/chainer/chainercv/blob/master/examples/fpn/train_multi.py#L98) to include `.astype(np.float32)` in order to work with the custom dataset (which is proved to be working with alternative models).

**Comment by yuyu2172** [comment]
> When attempted to perform training on a single GPU, I noticed that the model is loaded in the GPU memory, but its utilization is 0%, and there is a CPU core at 100%. As expected, the training is extremely slow.

The code should utilize GPUs for the computation of convolution.
Inside `FPN.prepare`, the CPU data is transfered to GPU if the model is on GPU.
We decided to use a custom converter due to the dynamic instance size of `bbox`.

**Comment by 23pointsNorth** [comment]
I did a bit more debugging and indeed, when the TransformedDataset is called, the data is converted from numpy to cupy. 

However, I've noticed there is big delay (~70-80 sec.) between the calling of the dataset `get_item` and `trainchain` calculations. I couldn't find where the issue is, I assume extra copying from gpu-cpu-gpu (?).

I did manage to fix the issue, as I had called `model.prepare` within the transform similar to the [`Faster-RCNN` demo](https://github.com/chainer/chainercv/blob/master/examples/faster_rcnn/train.py#L36). Removing it fixed it. Not sure how this issue would have arised from this.

I will close the issue.
